armi/reactor/blueprints/__init__.py

```diff
@@ -342,6 +344,33 @@ def _assignTypeNums(self):
                     if bDesign not in self.blockDesigns:
                         self.blockDesigns.add(bDesign)
 
+    def _autoUpdateNuclideFlags(self, cs):
+        """
+        This method is responsible for examining the fission product model treatment
+        that is selected by the user and adding a set of nuclides to the `nuclideFlags`
+        list.
+        Notes
+        -----
+        The reason for adding this method is that when switching between fission product
+        modeling treatments it can be time-consuming to manually adjust the `nuclideFlags`
+        inputs. This is specifically the case with the fission product model is set to
+        `explicitFissionProducts`.
+        """
+        nbs = lumpedFissionProduct.getAllNuclideBasesByLibrary(cs)
+        if nbs:
+            runLog.info(
+                f"Adding explicit fission products to the nuclide flags based on the "
+                f"fission product model set to `{cs['fpModel']}`."
+            )
+            for nb in nbs:
+                nuc = nb.name
+                if nuc in self.nuclideFlags or elements.byZ[nb.z] in self.nuclideFlags:
+                    continue
+                nuclideFlag = isotopicOptions.NuclideFlag(
+                    nuc, burn=False, xs=True, expandTo=[]
+                )
+                self.nuclideFlags[nuc] = nuclideFlag
+
```

Comment:
See comment above on why this exists. Basically, it's for user convenience and allows easy switching between fission product model options without requiring the user to constantly update the `nuclideFlags` in the blueprints.